cruise.umple/src/UmpleInternalParser_CodeClass.ump

```diff
@@ -2739,65 +2738,33 @@ class UmpleInternalParser
       ((UmpleTrait)uClassifier).addCodeInjection(injection);
     }
   }
-  
-  private boolean checkCodeInjectionValidityCustom(Token injectToken, UmpleClass uClass) {
-    List<Method> methods = uClass.getMethods();
-    String[] allOperations = getOperationName(injectToken).split(",");
-    List<String> allParameters = getOperationsParameters(injectToken);
 
-    for(int opInd = 0; opInd < allOperations.length; opInd++)
-    {
-      String operation = allOperations[opInd];
-      String currentParameters = allParameters.get(opInd);
-
-      Boolean matches = false;
-
-      for(Method method : methods) 
-      { 
-        boolean tmpMatches = false;
+  private boolean checkParameterMatch(List<MethodParameter> parameters, String currentParameters) {
+    // Check if method parameters match injection parameters
+    boolean isParameterMatch = true;
 
-        // Check if method matches operation
-        if (uClass.matchOperationMethod(operation, method.getName())) {
-          tmpMatches = true;
-        }
-      
-        // Check if method parameters match injection parameters
-        boolean isParameterMatch = true;
-        List<MethodParameter> parameters = method.getMethodParameters();
-
-        if("...".equals(currentParameters) || (parameters.size() == 0 && "".equals(currentParameters))) {
-          isParameterMatch = true;
-        }
-        else if(parameters.size() != currentParameters.split(",").length)
-        {
+    if("...".equals(currentParameters) || (parameters.size() == 0 && "".equals(currentParameters))) {
+      isParameterMatch = true;
+    }
+    else if(parameters.size() != currentParameters.split(",").length)
+    {
+      isParameterMatch = false;
+    }
+    else {
+      int indx = 0;
+      for(String parameterType : currentParameters.split(",")) {
+        if(!parameterType.equals(parameters.get(indx).getType())) {
           isParameterMatch = false;
         }
-        else {
-          int indx = 0;
-          for(String parameterType : currentParameters.split(",")) {
-            if(!parameterType.equals(parameters.get(indx).getType())) {
-              isParameterMatch = false;
-            }
-            indx++;
-          }
-        }
-
-        tmpMatches &= isParameterMatch;
-        matches = matches || tmpMatches;
-      }
-
-      if(!matches) {
-        getParseResult().addErrorMessage(new ErrorMessage(1012, injectToken.getPosition(), operation));
+        indx++;
       }
     }
-    
-    return false;
-  } 
 
-  private boolean checkCodeInjectionValidity(Token injectToken, UmpleClass uClass) {
-    ArrayList<String> methodNames = new ArrayList<String>();
-    String operationName = getOperationName(injectToken);
+    return isParameterMatch;
+  }
 
+  ArrayList<String> getAllMethodNames(UmpleClass uClass) {
```

Comment:
This should return `List<String>`. Always prefer Interfaces to concrete classes.